src/debug/HyprCtl.cpp

```diff
@@ -1821,7 +1824,17 @@ int hyprCtlFDTick(int fd, uint32_t mask, void* data) {
     if (isFollowUpRollingLogRequest(request)) {
         Debug::log(LOG, "Followup rollinglog request received. Starting thread to write to socket.");
         Debug::RollingLogFollow::Get().StartFor(ACCEPTEDCONNECTION);
-        runWritingDebugLogThread(ACCEPTEDCONNECTION);
+        std::istringstream iss(request);
+        std::string        val;
+        if (iss >> val) {
+            if (iss >> val) {
+                runWritingDebugLogThread(ACCEPTEDCONNECTION, std::stoi(val));
+            } else {
+                runWritingDebugLogThread(ACCEPTEDCONNECTION);
+            }
+        } else {
+            Debug::log(ERR, "Invalid rollinglog request");
+        }
```

Comment:
this construction does not look safe

is it working?

Reply:
yes